lib/index.js

```diff
@@ -31,10 +32,24 @@ export const httpClientHandler = {
       throw new Error('NotFoundError');
     }
 
-    return result.data;
+    return _getJsonData(result);
   }
 };
 
+async function _getJsonData(result) {
+  if(result.data) {
+    return result.data;
+  }
+
+  try {
+    return await result.json();
```

Comment:
I'm not sure this should be open ended and always try to parse arbitrary input.  One specific use case is when you get `application/octet-stream` due to the server not knowing what to do (ex, on github pages with no extension).  What other content types make sense to try and parse?  If it's HTML or an image, better to not bother and let it fail.  So maybe should refactor to something like as a special case:
```js
if(result.headers.get('content-type') === 'application/octet-stream') {
  // try the parse
} else {
  // throw an unsupported content type error
}
```
(And consider using `content-type` npm package for parsing the header string.)